inlong-sort/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/sink/multiple/WholeDatabaseMigrationOperator.java

```diff
@@ -0,0 +1,228 @@
+            MultipleSinkOption multipleSinkOption) {
+        this.catalogLoader = catalogLoader;
+        this.multipleSinkOption = multipleSinkOption;
+    }
+
+    @Override
+    public void open() throws Exception {
+        super.open();
+        this.catalog = catalogLoader.loadCatalog();
+        this.asNamespaceCatalog =
+                catalog instanceof SupportsNamespaces ? (SupportsNamespaces) catalog : null;
+        this.recordQueues = new HashMap<>();
+        this.schemaCache = new HashMap<>();
+        this.dynamicSchemaFormat = DynamicSchemaFormatFactory.getFormat(multipleSinkOption.getFormat());
+    }
+
+    @Override
+    public void close() throws Exception {
+        super.close();
+        if (catalog instanceof Closeable) {
+            ((Closeable) catalog).close();
+        }
+    }
+
+    @Override
+    public void processElement(StreamRecord<RowData> element) throws Exception {
+        String wholeData = element.getValue().getString(0).toString();
+
+        JsonNode jsonNode = objectMapper.readTree(wholeData);
+        boolean isDDL = dynamicSchemaFormat.extractDDLFlag(jsonNode);
+        if (isDDL) {
+            execDDL(jsonNode);
+        } else {
+            execDML(jsonNode);
+        }
+    }
+
+    private void execDDL(JsonNode jsonNode) {
+        // todo:parse ddl sql
+    }
+
+    private void execDML(JsonNode jsonNode) throws IOException {
+        RecordWithSchema record = parseRecord(jsonNode);
+        Schema schema = schemaCache.get(record.getTableId());
+        Schema dataSchema = record.getSchema();
+        recordQueues.compute(record.getTableId(), (k, v) -> {
+            if (v == null) {
+                v = new LinkedList<>();
+            }
+            v.add(record);
+            return v;
+        });
+
+        if (schema == null) {
+            handleTableCreateEventFromOperator(record.getTableId(), dataSchema);
+        } else {
+            handleSchemaInfoEvent(record.getTableId(), schema);
+        }
+    }
+
+
+    // ================================ 所有的与coordinator交互的request和response方法 ============================
+    private void handleSchemaInfoEvent(TableIdentifier tableId, Schema schema) {
+        schemaCache.put(tableId, schema);
+        Schema currentSchema = schemaCache.get(tableId);
+        Queue<RecordWithSchema> queue = recordQueues.get(tableId);
+        while (queue != null && !queue.isEmpty()) {
+            Schema dataSchema = queue.peek().getSchema();
+            // if compatible, this means that the current schema is the latest schema
+            // if not, prove the need to update the current schema
+            if (isCompatible(currentSchema, dataSchema)) {
+                RecordWithSchema recordWithSchema = queue.poll();
+                output.collect(new StreamRecord<>(
+                        recordWithSchema
+                                .refreshFieldId(currentSchema)
+                                .refreshRowData((jsonNode, schema1) ->

```

Comment:
Maybe it is better to call 'latestSchema' or 'validSchema'?

Reply:
ok， it is better than 'currentSchema'